src/averageAges.js

```diff
@@ -37,7 +42,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothersNames = [];
+  const women = people.filter(person => {
+    person.mother && mothersNames.push(person.mother);
+
+    return person.sex === 'f';
+  });
+  const womenWithChildren = withChildren && women.filter(
+    woman => mothersNames.includes(woman.name),
+  );
+  const selectedWomenList = withChildren ? womenWithChildren : women;
```

Comment:
do not duplicate check
```suggestion
  const womenWithChildren = women.filter(
    woman => mothersNames.includes(woman.name),
  );
  const selectedWomenList = withChildren ? womenWithChildren : women;
```